src/commands/Miscellaneous/discordstatus.ts

```diff
@@ -15,27 +15,121 @@ export default class extends SteveCommand {
+		const embedData = msg.language.tget('commandDiscordStatusEmbed');
+
+		const { data: currentStatus, statusText } = await axios.get<DiscordStatus>(url);
+
+		if (statusText !== 'OK') {
+			return msg.channel.send(msg.language.tget('commandDiscordStatusError'));
+		}
+
+		const fields: EmbedField[] = [];
+
+		if (currentStatus.components?.every(component => component.status === 'operational')) {
+			fields.push({
+				name: embedData.fields.operational.title,
+				value: embedData.fields.operational.value,
+				inline: false
+			});
+		} else {
+			currentStatus.components.forEach(component => {
+				if (component.status !== 'operational') {
+					fields.push({
 						name: component.name,
 						value: component.status,
 						inline: true
-					})))
-					.setFooter(embedData.footer(formatDate(json.page.updated_at)));
-
-				return msg.channel.send(embed);
-			})
-			.catch(err => {
-				console.log(err);
-				return msg.sendLocale('commandDiscordStatusError');
+					});
+				}
 			});
+		}
+
+		if (currentStatus.scheduled_maintenances.length > 0) {
+			fields.push({
+				name: embedData.fields.maintenance.title,
+				value: currentStatus.scheduled_maintenances.map(m => embedData.fields.maintenance.value(m.name, m.impact)).join('\n'),
+				inline: false
+			});
+		}
+
+		const embed = new MessageEmbed()
+			.setTitle(currentStatus.status.description)
+			.setDescription(embedData.description(currentStatus.incidents[0]
+				? currentStatus.incidents.map(i => i.name).join('\n')
+				: embedData.noIncidents))
+			.addFields(fields)
+			.setThumbnail('https://discord.com/assets/2c21aeda16de354ba5334551a883b481.png')
+			.setTimestamp()
+			.setFooter(embedData.footer(formatDate(new Date(currentStatus.page.updated_at))))
+			.setColor('BLURPLE');
+
+		return msg.channel.send(embed);
+
 	}
 
 }
+
+interface DiscordStatus {
+	page: Page;
+	status: Status;
+	components: ComponentsEntity[];
+	incidents: IncidentsEntity[];
+	scheduled_maintenances: ScheduledMaintenancesEntity[];
+}
+interface Page {
+	id: string;
+	name: string;
+	url: string;
+	updated_at: string;
+}
+interface Status {
+	description: string;
+	indicator: string;
+}
+interface ComponentsEntity {
+	created_at: string;
+	description?: string | null;
+	id: string;
+	name: string;
+	page_id: string;
+	position: number;
+	status: string;
+	updated_at: string;
+	only_show_if_degraded: boolean;
+}
+interface IncidentsEntity {
+	created_at: string;
+	id: string;
+	impact: string;
+	incident_updates?: IncidentUpdatesEntity[] | null;
+	monitoring_at?: null;
+	name: string;
+	page_id: string;
+	resolved_at?: null;
+	shortlink: string;
+	status: string;
+	updated_at: string;
+}
+interface IncidentUpdatesEntity {
+	body: string;
+	created_at: string;
+	display_at: string;
+	id: string;
+	incident_id: string;
+	status: string;
+	updated_at: string;
+}
+interface ScheduledMaintenancesEntity {
+	created_at: string;
+	id: string;
+	impact: string;
+	incident_updates?: IncidentUpdatesEntity[] | null;
```

Comment:
can this (and the rest of the optionals) actually be null *or* undefined? or is it gonna be one or the other?

Reply:
I'm not 100% sure. I used an interface generator one some of the examples to get the needed interfaces since there are frequently no scheduled maintenances or incidents. I could make it one or the others but then I'd have to wait for discord to break something to test it and I've already tested a version using these interfaces in a different project.

Reply:
aight, i'll see what's up next time discord breaks lol